hydromt_fiat/workflows/exposure_vector.py

```diff
@@ -538,52 +548,89 @@ def setup_ground_floor_height(
-        country: Union[str, None] = None,
-    ):
-        if max_potential_damage == "jrc_damage_values":
-            damage_source = self.data_catalog.get_dataframe(max_potential_damage)
-            if country is None:
-                country = "World"
-                self.logger.warning(
-                    f"No country specified, using the '{country}' JRC damage values."
+        max_potential_damage: Union[int, float, str, Path, List[str], List[Path], pd.DataFrame]=None,
+        target_attribute: Union[str, List[str], None] = None,
+        attr_name: Union[str, List[str], None] = None,
+        method: Union[str, List[str], None] = "nearest",
+        max_dist: float = 10,
+    ) -> None:
+        
+        if isinstance(max_potential_damage, pd.DataFrame
+            ):
+                self.update_max_potential_damage(
+                    updated_max_potential_damages=max_potential_damage
                 )
+        elif isinstance(max_potential_damage, int) or isinstance(
+            max_potential_damage, float
+        ):      
+                # modify the column manually 
+                #self.exposure_db[target_attribute] = max_potential_damage
+                NotImplemented
 
-            damage_values = preprocess_jrc_damage_values(damage_source, country)
-
-        elif max_potential_damage == "hazus_max_potential_damages":
-            damage_source = self.data_catalog.get_dataframe(max_potential_damage)
-            damage_values = preprocess_hazus_damage_values(damage_source)
-        else:
+        elif isinstance(max_potential_damage, str) or isinstance(
+            max_potential_damage, Path
+        ):
+            # A single file is used to assign the ground floor height to the assets
+            gfh = self.data_catalog.get_geodataframe(max_potential_damage)
+            gdf = self.get_full_gdf(self.exposure_db)
+            gdf = join_spatial_data(gdf, gfh, attr_name, method, max_dist, self.logger)
+            self.exposure_db = self._set_values_from_other_column(
+                gdf, target_attribute, attr_name
+            )
+        elif isinstance(max_potential_damage, list):
+            # Multiple files are used to assign the ground floor height to the assets
             NotImplemented
-            # TODO: Implement other ways of setting the max potential damage, e.g. from a
-            # vector file or from a table.
-
-        # Calculate the area of each object
-        gdf = self.get_full_gdf(self.exposure_db)[["Primary Object Type", "geometry"]]
-        gdf = get_area(gdf)
+            
 
-        # Set the damage values to the exposure data
-        if damage_types is None:
-            damage_types = ["total"]
+    def setup_ground_elevation(
+        self,
+        ground_elevation: Union[int, float, None, str, Path, List[str], List[Path]],
+        exposure_db: pd.DataFrame,
+        exposure_geoms: gpd.GeoDataFrame,
+    ) -> None:
 
-        for damage_type in damage_types:
```

Comment:
Lines 571-586 should also be included still I think. Please check how the `setup_max_potential_damage` function is now in the main branch and add your code then as other if/else options